### PR TITLE
[FIX] Prevent travel modal scroll to jump on autosave

### PR DESCRIPTION
**Pull request by AzaroQL**
# Description

Travel modal during autosave was scrolled to top.

Before:

https://user-images.githubusercontent.com/9656961/229831002-78e62418-02d9-4f89-ac9c-c0a5ddb509e3.mp4

After:

https://user-images.githubusercontent.com/9656961/229831060-5fc3ccee-03c4-4828-bdc5-00f4ca45def1.mp4

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

manual test + yarn test

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/expansion/components/travel/IslandList.tsx [hunk 1]
```diff
@@ -129,9 +125,14 @@ export const IslandList: React.FC<IslandListProps> = ({
   bumpkin,
   showVisitList,
   inventory,
+  travelAllowed,
 }) => {
   const { authService } = useContext(Auth.Context);
-  const [authState, send] = useActor(authService);
+  const farmId = useSelector(authService, (state) => state.context.farmId);
```
Comment (spencerdezartsmith) [inline src/features/game/expansion/components/travel/IslandList.tsx:131]: It is best practice to define the `useSelector` callbacks outside of the component as mentioned in the [docs](https://xstate.js.org/docs/packages/xstate-react/#api). This way the arrow function is not redefined on each rerender.

Would we be able to move these outside?

**Review by spencerdezartsmith — CHANGES_REQUESTED** [review]
Nice find/fix. Just one change mentioned.

File: src/features/game/expansion/components/travel/IslandList.tsx (hunk 1 above)
Comment (AzaroQL) [reply src/features/game/expansion/components/travel/IslandList.tsx]: @spencerdezartsmith, Done.

**Review by spencerdezartsmith — APPROVED** [review]
LGTM 🚀

**Pull request merged**